server/api/images.js

```diff
@@ -0,0 +1,20 @@
+// @flow
+import Router from 'koa-router';
+import auth from '../middlewares/authentication';
+import { Attachment } from '../models';
+import { getSignedImageUrl } from '../utils/s3';
+
+const router = new Router();
+
+router.post('images.info', auth(), async ctx => {
+  const { id } = ctx.body;
+  ctx.assertPresent(id, 'id is required');
+  const attachment = await Attachment.findByPk(id);
+  const redirectUrl =
+    attachment.acl === 'private'
+      ? await getSignedImageUrl(attachment.key)
+      : attachment.url;
+  ctx.redirect(redirectUrl);
+});
```

Comment:
three points that I can think of here:
1) not major at the moment, but can this become a more generic `attachments.info` endpoint, as it does not have to just apply to image proxies now?
2) permissions: at the moment no checks happening, it is all passive. i.e. if you are viewing the document then you can get to the images. Where if any should some checks go?
3) shared documents. I think I just totally missed this use-case. Lowest effort outcome, if you are using `AWS_S3_ACL=private` then essentially you can't share publicly. Highest effort outcome, similar to the `documents.info` api, we could do some share checks and remove the `auth()` requirement.

Reply:
1. Agree, how about `attachments.redirect`? We should reserve `info` for a json response with the attachment details.

2. After we query the attachment model on L12, query it's document with the current user, something like:

```
const user = ctx.state.user;
const document = await Document.findByPk(id, { userId: user.id });
authorize(user, 'read', document);
```

I think for now we can say that if you have access to view the document then you have access to read it's attachments without creating a new, separate authentication policy for attachments specifically. 

3. Public shares do complicate things a little. I'm happy to take this one if you like – my current gut is to have `presentDocument` (the document presenter) rewrite the text contents with the signed urls before it sends them down to the client if the document is public.

Reply:
1) done as per suggestion
2) done as per suggestion
3) I will take your offer of help! Will wrap up anything else you need for this PR and get it ready for merging